### DISCUSSION TITLE
Social network API integrated with python

### DISCUSSION THREAD

**Discussion by ThiagoTemoteo** [community/community#123539] [Programming Help] [answered]
### Body

I'm a technical student and I need help with programming in Python. My professor gave me a project in which I must use an API from some social network and integrated with Python. My group's idea is to capture data such as the number of shares, likes, reposts and perhaps some other information. I thought about Instagram and went after it to see if I could find a solution. However, I didn't find any videos on the internet and the official API guide on the Meta Developer website is very confusing for me. Can anyone tell me if this project is viable and if it can go ahead?

Note: I have 3 weeks difference from today.

### Guidelines

- [ ] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by davevad93** [comment, marked as answer]
Hi @thibasBigT ,

I've never personally used Instagram API but I think 3 weeks are enough if your assignment consists in scraping data from a social media with Python, here below I'm gonna show a few links that you might find useful for your task:

- [A Beginner’s Guide to Interacting with Instagram Using Python and the Graph API](https://ensembledata.medium.com/using-python-to-scrape-data-from-instagram-89b0a5122143)
- [Using Python to scrape data from Instagram](https://ensembledata.medium.com/using-python-to-scrape-data-from-instagram-89b0a5122143)
- [How to Fetch Data From Instagram Using Python](https://www.makeuseof.com/python-instagram-fetch-data/)

**Reply by ThiagoTemoteo** [reply]
Thank you very much for the tip, I was really looking for an alternative because I'm looking at their terms and it seems that on Instagram you need to already have some well-developed application with a really useful purpose and that will "appropriately use" their data, something that not even close I have XD

**Reply by davevad93** [reply]
You're welcome and good luck with your task!

**Reply by ThiagoTemoteo** [reply]
I used EnsembleData and everything actually went well. I managed to meet my teacher’s expectations and deliver the project. However, I was left with a doubt that I couldn’t understand why this happened.

```
import requests

root = "https://www.ensembledata.com/apis"
endpoint = "/youtube/channel/detailed-info"
params = {
  "browseId": "UCnQghMm3Z164JFhScQYFTBw",
  "from_url": False,
  "token": "XXXXXXXXXXXXX"
}

res = requests.get(root+endpoint, params=params)
data = res.json()

for dados in data['data']['metadata']['aboutChannelViewModel']:
  print(f"Descrião do canal pesquisado: {dados['description']}")
  print(f"Numero de inscritos no canal: {dados['subscriberCountText']}")
  print(f"Numero de vizualizações acumuladas neste canal: {dados}['viewCountText']")
  print(f"Nome do canal pesquisado:")
  print(f"URL do canal pesquisado: {dados}['displayCanonicalChannelUrl']")
```
  
  When I ran this code, it gave the following error:
  
  PS C:\Users\thiba\OneDrive\Documents\Códigos> & C:/Users/thiba/AppData/Local/Programs/Python/Python312/python.exe c:/Users/thiba/OneDrive/Documents/Códigos/youtube1.py
Traceback (most recent call last):
  File "c:\Users\thiba\OneDrive\Documents\Códigos\youtube1.py", line 15, in <module>
    print(f"Descrião do canal pesquisado: {dados['description']}")
                                           ~~~~~^^^^^^^^^^^^^^^
TypeError: string indices must be integers, not 'str'
PS C:\Users\thiba\OneDrive\Documents\Códigos> 

and this only corrected itself when I removed the FOR loop from my code and left it ending like this:

```
_res = requests.get(root+endpoint, params=params)
data = res.json()

dados = data['data']['metadata']['aboutChannelViewModel']

print(f"Descrição do canal pesquisado: {dados['description']}")
print(f"Numero de inscritos no canal: {dados['subscriberCountText']}")
print(f"Numero de visualizações acumuladas neste canal: {dados['viewCountText']}")
print(f"URL do canal pesquisado: {dados['displayCanonicalChannelUrl']}")_

```

Why did this happen? I made it work without having much idea of how I made it work LOL.

**Reply by davevad93** [reply]
Hey @thibasBigT , I'm really glad that your task went well. About the error `TypeError: string indices must be integers, not 'str'` it could be for 2 reasons:

1. Because using a `for-loop` over a dictionary iterates over its keys, not its values, so iterating over it to access the values raises that error. 

2. Trying to access an element as a `list` when it's actually a `dictionary`,

Basically, the approach of the second code is based on accessing the dictionary directly without iterating over it.

**Comment by ebndev** [comment]
Would Spotify count as a social network platform? I found their APIs easy to use and you can fetch data from user profiles like their top played songs, followed artists, etc.

Here's the [documentation](https://developer.spotify.com/documentation/web-api/reference/get-current-users-profile).

**Comment by judahpaul16** [comment]
Check out the [Social Blade API](https://socialblade.com/business-api). Also, [Postman ](https://www.postman.com/)is a good tool to have regardless of what API you're testing. I'm not sure what pricing looks like for Social Blade's API or if there's a free tier but there's also good old fashion [web scraping](https://medium.com/swlh/how-to-scrape-socialblade-for-youtube-subscription-data-ec7c4bde6933) with `beautifulsoup` and `selenium` if all else fails. Here's [an example](https://github.com/judahpaul16/social-media-analytics/) I made a while back.

**Reply by ebndev** [reply]
That's awesome @judahpaul16! Thank you for sharing those resources and your repo 😄
